From 59ac5dd8761bcdd7128ca3195b1c4920b9213539 Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Sun, 8 Jun 2014 07:33:29 +0530
Subject: [PATCH 18/39] ASoC: soc-fw: move control header inside the loop

Issue: GMINL-3679

the mixer control indexes can be different as each
mixer will have a separate control header so move the
header to inside the loop.

Change-Id: I9e22dbc223933a499a892a3ad4c7ef7ec39296a2
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/soc-fw.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sound/soc/soc-fw.c b/sound/soc/soc-fw.c
index 92bea0a..2dece11 100644
--- a/sound/soc/soc-fw.c
+++ b/sound/soc/soc-fw.c
@@ -804,11 +804,13 @@ static int soc_fw_kcontrol_load(struct soc_fw *sfw, struct snd_soc_fw_hdr *hdr)
 	}
 
 	sfw->pos += sizeof(struct snd_soc_fw_kcontrol);
-	control_hdr = (struct snd_soc_fw_control_hdr *)sfw->pos;
 
 	dev_dbg(sfw->dev, "ASoC: adding %d kcontrols\n", sfwk->count);
 
 	for (i = 0; i < sfwk->count; i++) {
+
+		control_hdr = (struct snd_soc_fw_control_hdr *)sfw->pos;
+
 		switch (SOC_CONTROL_GET_ID_INFO(control_hdr->index)) {
 		case SOC_CONTROL_TYPE_VOLSW:
 		case SOC_CONTROL_TYPE_STROBE:
-- 
1.9.1

